src/app/tabs/run-tab.js

```diff
@@ -239,6 +240,84 @@ function updateAccountBalances (container, appAPI) {
 }
 
 /* ------------------------------------------------
+    RECORDER
+------------------------------------------------ */
+function makeRecorder (appAPI, appEvents) {
+  var udapp = appAPI.udapp()
+  var recorder = new Recorder({
+    events: {
+      udapp: appEvents.udapp,
+      executioncontext: executionContext.event
+    },
+    api: {
+      getAccounts (cb) { udapp.getAccounts(cb) }
+    }
+  })
+  var css = csjs`
+    .container {
+      margin-top: 10px;
+      display: flex;
+    }
+    .recorder {
+      ${styles.button}
+      width: 135px;
+    }
+    .runTxs {
+      ${styles.button}
+      margin-left: 10px;
+      width: 135px;
+    }
+  `
+  var recordButton = yo`<button class=${css.recorder}>copy ran transactions</button>`
+  var runButton = yo`<button class=${css.runTxs}>re-run transactions</button>`
+  var el = yo`
+    <div class=${css.container}>
+     ${recordButton}
+     ${runButton}
+    </div>
+  `
+  recordButton.onclick = () => {
+    var txJSON = JSON.stringify(recorder.getAll(), null, 2)
+    copy(txJSON)
+    modalDialogCustom.alert(txJSON)
+  }
+  runButton.onclick = () => {
+    var opts = { title: `Enter Transactions`, text: `Paste the array of transaction you want to replay here`, inputValue: '', multiline: true }
+    modalDialogCustom.prompt(opts, function confirm (json = '[]') {
+      try {
+        var txArray = JSON.parse(json)
+      } catch (e) {
+        modalDialogCustom.alert('Invalid JSON, please try again')
+      }
+      if (txArray.length) {
+        txArray.forEach(tx => udapp.replayTx(tx.record, CALLBACK))
+      }
+    }, function cancel () { })
+  }
+  function CALLBACK (err, result) {
+    if (err) console.error(err)
+    else {
+      console.log(result)
+      // {
+      //   "result": {
+      //     "gasUsed": "5318",
+      //     "vm": { "exception": 1, "selfdestruct": {} },
+      //     "bloom": { "bitvector": { "type": "Buffer", "data": [0, /* ... */ 0, 0] } },
+      //     "amountSpent": "5318"
+      //   },
+      //   "transactionHash": "0x84f68f96944a47b27af4b4ed1986637aa1bc05fd7a6f5cb1d6a53f68058276d8"
+      // }
+      /*
+        at each callback call, if the transaction succeed and if this is a creation transaction, we should call
+
+        runtab.addInstance( ... ) // which basically does:
+        instanceContainer.appendChild(appAPI.udapp().renderInstance(contract, address, selectContractNames.value))
```

Comment:
at this point if we have a contract creation we should add a new instance

Reply:
ok. but i guess the "op" bug needs to be fixed to test this